bb8/src/internals.rs

```diff
@@ -273,12 +319,30 @@ impl From<&AtomicStatistics> for Statistics {
             get_direct: item.get_direct.load(Ordering::SeqCst),
             get_waited: item.get_waited.load(Ordering::SeqCst),
             get_timed_out: item.get_timed_out.load(Ordering::SeqCst),
+            get_waited_time: Duration::from_micros(
+                item.get_waited_time_micros.load(Ordering::SeqCst),
+            ),
+            connections_created: item.connections_created.load(Ordering::SeqCst),
+            connections_broken_closed: item.connections_broken_closed.load(Ordering::SeqCst),
+            connections_invalid_closed: item.connections_invalid_closed.load(Ordering::SeqCst),
+            connections_max_lifetime_closed: item
+                .connections_max_lifetime_closed
+                .load(Ordering::SeqCst),
+            connections_max_idle_timeout_closed: item
+                .connections_max_idle_timeout_closed
+                .load(Ordering::SeqCst),
         }
     }
 }
 
 pub(crate) enum StatsKind {
-    Direct,
-    Waited,
-    TimedOut,
+    GetDirect,
```

Comment:
I think we'll want to split this into `StatsGet { Waited, Direct, TimedOut }`.

Reply:
@djc just for having an agreement on the new kinds and record operations that we would have within `AtomicStatistics`, we can go for this

```rust
pub(crate) enum StatsGet {
    Waited,
    Direct,
    TimedOut,
}

pub(crate) enum StatsConnectionClosed {
    Broken,
    Invalid,
}
```

Note: we should not need more stats kind since the used `record_*` methods for stats like the ones coming from the reap function will just call an ad-hoc function like `record_connections_reaped`

And then we would have the following `record_*` methods within the `AtomicStatistics`:

- `record_get` (takes the `StatsGet` parameter)
- `record_connection_closed`  (takes the `StatsConnectionClosed` parameter)
- `record_connections_reaped`
- `record_get_wait_time` 
- `record_connection_created`

Let me know if this works for you and i can create a new PR with separated commits for each group, starting with the rename of the `record` to `record_get`

Reply:
No, as suggested in my previous message we should reuse a simple `record()` with `StatsKind` for both `connection_closed` and `connection_created`, and `record_get()` should also handle `get_wait_time`.

Reply:
Oks, let me provide a new draft - sorry for the back and forth here :

```
pub(crate) enum StatsGet {
    Waited,
    Direct,
    TimedOut,
    WaitTime
}

pub(crate) enum StatsConnectionClosed {
    Broken,
    Invalid,
}
```

And following `record` methods

- `record` which would take a `StatsConnectionClosed` kind type
- `record_get` for recording any stats related to get including wait time
- `record_connections_reaped` for recording the reaped connections either idle timeout or max lifetime

How does it sound? is this aligned with your expectations?

Reply:
No. Let's do one PR at a time, please implement the first chunk and I'll give feedback. Please closely read my feedback from this PR because you've still missed some of it.